js/frames/versionSaga.js

```diff
@@ -110,13 +110,40 @@ function sagaEdited() {
 
 function updateAbilityHeights() {
 	const maxHeight = card.text.type.y - card.text.ability0.y;
-	const height = maxHeight / card.saga.count;
-	card.text.ability0.height = 0
-	card.text.ability1.height = 0
-	card.text.ability2.height = 0
-	card.text.ability2.height = 0
+	
+	// Get all saga abilities that have content
+	const abilities = [];
 	for (let i = 0; i < card.saga.count; i++) {
-		card.text[`ability${i}`].height = height;
+	const abilityText = card.text[`ability${i}`].text;
+	// Count words excluding reminder text in parentheses
+	const wordCount = abilityText
+		.replace(/\([^)]*\)/g, '') // Remove reminder text
+		.trim()
+		.split(/\s+/)
+		.length;
+		
+	abilities.push({
+		index: i,
+		wordCount: wordCount
+	});
+	}
+
+	// Calculate proportional heights
+	const totalWords = abilities.reduce((sum, a) => sum + a.wordCount, 0);
+	
+	// Add height constraint (minimum height)
+	const minHeight = maxHeight * 0.15; // 15% of max height
+	abilities.forEach(ability => {
+	const height = Math.max(
+		minHeight,
+		(ability.wordCount / totalWords) * maxHeight
+	);
+	card.text[`ability${ability.index}`].height = height;
```

Comment:
Because of the `minHeight` exception I think this won't work correctly for short ones? Going to try a more complex algorithm here. I think we need to reserve the minimum and then redistribute the rest of the space or something like that?